client/layout/guided-tours/config-elements/step.tsx

```diff
@@ -183,29 +177,36 @@ export default class Step extends Component< Props, State > {
 	}
 
 	watchTarget() {
+		const { target, onTargetDisappear, waitForTarget, keepRepositioning } = this.props;
 		if (
-			! this.props.target ||
-			! this.props.onTargetDisappear ||
+			( ! keepRepositioning && ( ! target || ( ! onTargetDisappear && ! waitForTarget ) ) ) ||
 			typeof window === 'undefined' ||
 			typeof window.MutationObserver === 'undefined'
 		) {
 			return;
 		}
 
+		this.safeSetState( { seenTarget: ! target || targetForSlug( target ) } );
+
 		if ( ! this.observer ) {
 			this.observer = new window.MutationObserver( () => {
-				const { target, onTargetDisappear } = this.props;
+				if ( keepRepositioning ) {
+					this.onScrollOrResize();
+				}
 
-				if ( ! target || ! onTargetDisappear ) {
+				if ( ! target || ( ! waitForTarget && ! onTargetDisappear ) ) {
```

Comment:
This conditional looks quite complex, it might be nice to comment it to explain what it does (or move it to a function).